src/JuMP.jl

```diff
@@ -438,6 +462,35 @@ end
 
 set_optimize_hook(model::Model, f) = (model.optimize_hook = f)
 
+"""
+    set_silent(model::Model)
+
+Takes precedence over any other attribute controlling verbosity 
+and requires the solver to produce no output.
+"""
+function set_silent(model::Model)
+    if mode(model) != DIRECT &&
+        MOIU.state(backend(model)) == MOIU.NO_OPTIMIZER
+        return "No optimizer attached."
```

Comment:
If no optimizer is attached, the silent attribute will be copied to the optimizer when it is attached so I don't think we need to make an exception

Reply:
Ok, I thought it should follow the same logic as the solver_name function, but indeed it makes sense that the silent works even if no optimizer is attached

```julia
function solver_name(model::Model)
    if mode(model) != DIRECT &&
        MOIU.state(backend(model)) == MOIU.NO_OPTIMIZER
        return "No optimizer attached."
    else
        return _try_get_solver_name(backend(model))
    end
end
```

Reply:
For what I understood now this is what we want to work
```julia
model = Model()
MOI.set(model, MOI.Silent(), true)
```
but unfortunately it gives this error
```julia
ERROR: MethodError: no method matching set(::Model, ::MathOptInterface.Silent, ::Bool)
Closest candidates are:
  set(::Model, ::MathOptInterface.AbstractModelAttribute, ::Any) at /Users/guilhermebodin/Documents/JuMP.jl/src/JuMP.jl:643
  set(::MathOptInterface.Utilities.MockOptimizer, ::Union{AbstractModelAttribute, AbstractOptimizerAttribute}, ::Any) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/mockoptimizer.jl:157
  set(::MathOptInterface.Utilities.CachingOptimizer, ::MathOptInterface.AbstractOptimizerAttribute, ::Any) at /Users/guilhermebodin/.julia/packages/MathOptInterface/C3lip/src/Utilities/cachingoptimizer.jl:511
```
Because `Silent` attribute is an `AbstractOptimizerAttribute`
```julia
struct Silent <: AbstractOptimizerAttribute end
```
I tested implementing the following function and now the first code works
```julia
MOI.set(m::Model, attr::MOI.AbstractOptimizerAttribute, value) = MOI.set(backend(m), attr, value)
model = Model()
MOI.set(model, MOI.Silent(), true)
```
and then I can write the `set_silent` as 

```julia
function set_silent(model::Model)
    return MOI.set(model, MOI.Silent(), true)
end
```
have I got it right?

Reply:
Yes, that looks correct :)